### PR TITLE
Adding index page for Azure Machine Learning Component

### PR DESCRIPTION
**Pull request by yilun-msft**
Adding an index page for Azure Machine Learning Component to provide capability to use Azure Machine Learning Service to register and deploy.

### CODE REVIEW

**Comment by yilun-msft** [comment]
@shbijlan if you could take a quick look before get reviews to set up a frame for our Azure component index page. Thanks a lot

**Comment by shbijlan** [comment]
@yilun-msft I have read through the document. It looks good. My only question is was it intentional to have the same two links to github directory (aml-register-model) and Model Register/Deploy?

**Comment by yilun-msft** [comment]
> @yilun-msft I have read through the document. It looks good. My only question is was it intentional to have the same two links to github directory (aml-register-model) and Model Register/Deploy?

The thing is that because the AML-deploy-model hasn't been checked in, and it will lead us to 404 error. Let me change it to aml-deploy-model page. I think it is safe that the backend code will be merged earlier than this website page change. Thanks!

**Comment by yilun-msft** [comment]
Metacomment: please hold this PR before the backend change merges into pipeline/component repo.

**Comment by 8bitmp3** [comment]
@yilun-msft I think you can use `do-not-merge/hold` https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md#the-code-review-process Let me try this to hold the PR:

do-not-merge/hold

**Comment by yilun-msft** [comment]
> @yilun-msft I think you can use `do-not-merge/hold` https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md#the-code-review-process Let me try this to hold the PR:
> 
> do-not-merge/hold

Thanks for the advice. The PR is ready to review. The only thing is that I would like to check-in this PR first https://github.com/kubeflow/pipelines/pull/4711 for the component itself.

File: content/en/docs/azure/machinelearningcomponent.md
Comment (8bitmp3) [inline content/en/docs/azure/machinelearningcomponent.md:3]: Suggestions:

- Improve the description:
  - The guide is about Azure ML components (plural) not one component
  - There is a typo ("on step in the ML workflow" instead of "one step" and I think the description describes a component, not what the guide does
  - I propose to add "overview" since there're links to Azure docs that contain more info about each component.

Let's try this:

```suggestion
description = "Overview of Azure Machine Learning pipeline components for workflow improvements"
```

(or something like that)

File: content/en/docs/azure/machinelearningcomponent.md
Comment (8bitmp3) [inline content/en/docs/azure/machinelearningcomponent.md:7]: Suggestions:

- Since you're using "Azure ML" in section titles (instead of the full name), let's define it
- Use sentence case (https://docs.microsoft.com/en-us/style-guide/top-10-tips-style-voice#when-in-doubt-dont-capitalize and https://docs.microsoft.com/en-us/style-guide/capitalization)
- Start with talking about Azure ML components (as a suite of products) and then describe what a component is
- "Lifecycle" (one word - https://docs.microsoft.com/en-us/search/?scope=Style%20guide&terms=lifecycle)
- Define ML before using "ML"

```suggestion
Azure Machine Learning (Azure ML) components are pipeline components that integrate with [Azure ML](https://docs.microsoft.com/en-us/azure/machine-learning/) to manage the lifecycle of your machine learning (ML) models to improve the quality and consistency of your machine learning solution. A pipeline component is a self-contained set of code that performs one step in the ML workflow. A component is analogous to a function, in that it has a name, parameters, return value and a body. 
```

File: content/en/docs/azure/machinelearningcomponent.md
Comment (8bitmp3) [reply content/en/docs/azure/machinelearningcomponent.md:3]: And let me know if "pipeline components" should be capitalized, if it's some sort of branded product suite at Azure. Then we can change it back to "Pipeline Components".

File: content/en/docs/azure/machinelearningcomponent.md
Comment (8bitmp3) [inline content/en/docs/azure/machinelearningcomponent.md:32]:
```suggestion
## Azure ML Register Model component

```

File: content/en/docs/azure/machinelearningcomponent.md
Comment (8bitmp3) [inline content/en/docs/azure/machinelearningcomponent.md:36]: Suggested refactoring:

```suggestion
To learn more about the Azure ML Register Model pipeline component, refer to the [official repository](https://github.com/kubeflow/pipelines/tree/master/components/azure/azureml/aml-register-model).

```

File: content/en/docs/azure/machinelearningcomponent.md
Comment (8bitmp3) [inline content/en/docs/azure/machinelearningcomponent.md:38]: - I think _"See more for Model Deploy on .. Page"_ may not be very helpful and descriptive to someone who isn't familiar with the platform. 
- The doc you're referring to here already describes what it does in the first sentence, so maybe you can use it almost word-for-word (it says "In this article, learn about how to use Azure Machine Learning to manage the lifecycle of your models...")
- Also, every web page is a page, so maybe we should use specific words, such as "guide" or "tutorial" or "documentation" or nothing at all.
- Suggested changes:

```suggestion
To learn more about using Azure ML to manage the lifecycle of your models, go to [Model management, deployment, and monitoring](https://docs.microsoft.com/en-us/azure/machine-learning/concept-model-management-and-deployment).
```

**Review by 8bitmp3 — CHANGES_REQUESTED** [review]
Hi @yilun-msft Thanks for this guide! Awesome work.

Below are my suggestions to help improve the doc (feel free to use your own cc @sudivate):

---

- Improve the description:
  - The guide is about Azure ML components (plural) not one component
  - There is a typo ("on step in the ML workflow" instead of "one step" and I think the description describes a component, not what the guide does
  - I propose to add "overview" since there're links to Azure docs that contain more info about each component.

Let's try this:

```suggestion
description = "Overview of Azure Machine Learning pipeline components for workflow improvements"
```

(or something like that)

---

- Since you're using "Azure ML" in section titles (instead of the full name), let's define it
- Use sentence case (https://docs.microsoft.com/en-us/style-guide/top-10-tips-style-voice#when-in-doubt-dont-capitalize and https://docs.microsoft.com/en-us/style-guide/capitalization)
- Start with talking about Azure ML components (as a suite of products) and then describe what a component is
- "Lifecycle" (one word - https://docs.microsoft.com/en-us/search/?scope=Style%20guide&terms=lifecycle)
- Define ML before using "ML"

```suggestion
Azure Machine Learning (Azure ML) components are pipeline components that 
integrate with [Azure ML](https://docs.microsoft.com/en-us/azure/machine-learning/) 
to manage the lifecycle of your machine learning (ML) models to improve the quality 
and consistency of your machine learning solution. A pipeline component is a 
self-contained set of code that performs one step in the ML workflow. A component 
is analogous to a function, in that it has a name, parameters, return value and a body. 
```

---

- Check the grammar: "components ... provide"
- Use active voice - "you can use .. components" vs "components are used" if you can - this guide can help the reader do X with Y
- Unless "Components" is an Azure product name, use sentence case (Microsoft (and Google) style guide: 
  - When in doubt, don't capitalize (Microsoft: https://docs.microsoft.com/en-us/style-guide/top-10-tips-style-voice#when-in-doubt-dont-capitalize)
  - "everything is lowercase except the first word and proper nouns, which include the names of brands, products, and services. https://docs.microsoft.com/en-us/style-guide/capitalization)
- Use bullet points as a visual aid if you can

```suggestion
You can use Azure Machine Learning components to increase the efficiency of your 
workflow with Azure Machine Learning, such as continuous integration, delivery, and 
deployment. 

The components provide capabilities for: 

- Faster experimentation and development of models
- Faster deployment of models into production
- Quality assurance
```

---

And let me know if "pipeline components" should be capitalized, if it's some sort of branded product suite at Azure. Then we can change it back to "Pipeline Components".

---

- Let's improve this sentence - it currently says _"- Follow the [instruction](https://www.kubeflow.org/docs/azure/) to install Kubeflow on AKS cluster."_ - there are 5 guides with nested guides in the provided link, so the readers may be confused about what to follow:

image

- Let's try this:

```suggestion
- You should have Kubeflow installed on your AKS cluster. If you don't, follow the 
[Kubeflow installation (for Azure) guide](https://www.kubeflow.org/docs/azure/deploy/install-kubeflow/).
```

---

- Fix a typo ("resources" not "resourses")
- Suggest to use "README" <- "readme.md"
- For accessibility, "refer to"/"check"/"go to" instead of "see" (a11y)
- "Check... to learn about" <- "See... to see" (repetition)

```suggestion
- To interact with Azure resources, you may need to configure them before using a 
particular pipeline component. Check the README for each component to learn about 
what Azure resources are required.
```

---

- Fix typo "extension" <- "extention"
- Add "a" (indefinite article) to "secret"
- Add "the" (definite article) to "`kfp.azure extension"

```suggestion
- The `kfp.azure` extension can be used to create a secret to interact with Azure 
resources. To create Azure credentials, run:
```

---

- Use sentence case (Microsoft (and Google) style guide: "everything is lowercase except the first word and proper nouns, which include the names of brands, products, and services... capitalize the first word of a sentence, heading, title, UI label (such as the name of a button or check box), or standalone phrase.") https://docs.microsoft.com/en-us/style-guide/capitalization)

```suggestion
## Azure ML Deploy Model component
```
```suggestion
## Azure ML Register Model component
```

---

- I think _"See more for Model Deploy on .. Page"_ may not be very helpful and descriptive to someone who isn't familiar with the platform. 
- The doc you're referring to here already describes what it does in the first sentence, so maybe you can use it almost word-for-word (it says "In this article, learn about how to use Azure Machine Learning to manage the lifecycle of your models...")
- Also, every web page is a page, so maybe we should use specific words, such as "guide" or "tutorial" or "documentation" or nothing at all.
- Suggested changes:

```suggestion
To learn more about using Azure ML to manage the lifecycle of your models, 
go to 
[Model management, deployment, and monitoring](https://docs.microsoft.com/en-us/azure/machine-learning/concept-model-management-and-deployment).
```

---

- I think you should specify which component - its name - in the first introductory sentence (below)
- Use definite/indefinite articles
- Use bullet points as a visual aid if you can

```suggestion
Trained machine learning models are deployed as web services in the cloud and 
you can use your model by accessing its endpoint. When using the model as a web 
service, the following items are included in the Azure ML Deploy Model component: 

- An entry script
- Azure ML environment configurations
```

---

- Do you know if the link is being added in another PR since at the moment it's not working? (below)
- Let's be descriptive in this sentence to help the reader navigate to the page, if they're looking for more details:

```suggestion
For more information about the Azure ML Deploy Model pipeline component, 
check the [official repository](https://github.com/kubeflow/pipelines/tree/master/components/azure/azureml/aml-deploy-model).
```

---

- Use sentence case (Microsoft (and Google) style guide: "everything is lowercase except the first word and proper nouns, which include the names of brands, products, and services... capitalize the first word of a sentence, heading, title, UI label (such as the name of a button or check box), or standalone phrase.") https://docs.microsoft.com/en-us/style-guide/capitalization)
- Add a blank line to separate the section title from the first paragraph
- a11y: "learn"/"navigate"/"go"/"refer" <- "see"

```suggestion
## Other Azure ML capabilities

You can learn more about the capabilities of Azure ML and how to improve 
your ML workflow by checking the [Azure ML documentation](https://docs.microsoft.com/en-us/azure/machine-learning/) 
```
---

Thank you again! 👍

**Comment by 8bitmp3** [comment]
Thanks for your efforts @yilun-msft ! 👋 Appreciate it

PTAL

/ok-to-test

/lgtm

/assign @aronchick @eedorenko @sudivate

**Pull request merged**
